### PR TITLE
Hexagon benchmarks new schedules

### PR DESCRIPTION
**Pull request by pranavb-ca**
Before:
Done, time (conv3x3a16): 0.0049175 s (128 byte mode)
Done, time (dilate3x3): 0.0042597 s (128 byte mode)
Done, time (median3x3): 0.0050552 s (128 byte mode)
Done, time (gaussian5x5): 0.0042595 s (128 byte mode)
Done, time (sobel): 0.0054079 s (128 byte mode)
Done, time (conv3x3a32): 0.0051582 s (128 byte mode)

After:
Done, time (conv3x3a16): 0.0017923 s (128 byte mode)
Done, time (dilate3x3): 0.0019679 s (128 byte mode)
Done, time (median3x3): 0.0021045 s (128 byte mode)
Done, time (gaussian5x5): 0.0021509 s (128 byte mode)
Done, time (sobel): 0.0019982 s (128 byte mode)
Done, time (conv3x3a32): 0.0021427 s (128 byte mode)

@dsharletg, @dpalermo, @ronlieb - There may be better schedules out there that we can explore over time. These are the best I have *so far*.

### CODE REVIEW

**Comment by ronlieb** [comment]
LATM

Looks Awesome To Me

**Comment by pranavb-ca** [comment]
I tried by adding `.store_root()` by that in fact degraded performance slightly
Done, time (conv3x3a16): 0.0017827 s (128 byte mode)
Done, time (dilate3x3): 0.0021886 s (128 byte mode)
Done, time (median3x3): 0.0023665 s (128 byte mode)
Done, time (gaussian5x5): 0.0023958 s (128 byte mode)
Done, time (sobel): 0.0021328 s (128 byte mode)
Done, time (conv3x3a32): 0.0021045 s (128 byte mode)

I did not `fold_storage` though. I am trying that right now.

**Comment by pranavb-ca** [comment]
`fold_storage` didn't help much either. I suggest we stick to these schedules.

File: apps/hexagon_benchmarks/conv3x3_generator.cpp [hunk 1]
```diff
@@ -39,7 +39,11 @@ class Conv3x3 : public Generator<Conv3x3> {
 
             Expr output_stride = output.dim(1).stride();
             output.dim(1).set_stride((output_stride/vector_size) * vector_size);
-            bounded_input.compute_root();
+            bounded_input
+                .compute_at(Func(output), y)
+                .store_at(Func(output), y)
```
Comment (dsharletg) [inline apps/hexagon_benchmarks/conv3x3_generator.cpp]: Remove this redundant store_at (it's the same as the default).

**Comment by pranavb-ca** [comment]
@dsharletg: PTAL.

**Comment by ronlieb** [comment]
The buildbots seemed to mostly pass, one failed building llvm.
There is no way this patch broke the build of llvm.

ok to merge ?

**Pull request merged**